sam_learning/learners/sam_learning.py

```diff
@@ -218,6 +223,44 @@ def deduce_initial_inequality_preconditions(self) -> None:
                 if action_data.signature[lifted_param1] == action_data.signature[lifted_param2]:
                     action_data.preconditions.root.inequality_preconditions.add((lifted_param1, lifted_param2))
 
+    def hard_remove_negative_preconditions(self):
+        """Removes all negative preconditions"""
+        for action in self.partial_domain.actions.values():
+            new_preconditions = set()
+
+            for precondition in action.preconditions.root.operands:
+                if isinstance(precondition, Predicate) and not precondition.is_positive:
+                    continue
+
+                new_preconditions.add(precondition)
+
+            action.preconditions.root.operands = new_preconditions
+
+    def soft_remove_negative_preconditions(self):
+        """Removes negative preconditions when the fluent is an add effect"""
+        for action in self.partial_domain.actions.values():
+            new_preconditions = set()
+            action_add_effects = [effect.untyped_representation for effect in action.discrete_effects if effect.is_positive]
+
+            for precondition in action.preconditions.root.operands:
+                if isinstance(precondition, Predicate) and not precondition.is_positive:
+                    copy_precondition_positive = precondition.copy()
```

Comment:
I added to pddl-plus-parser==3.6.3 the option to call copy as follows: 
result = predicate.copy(is_negated=True)
and then you can remove the extra line where you set the is_positive value.